unitTests/ui/DidYouMeanTest.ts

```diff
@@ -65,61 +65,89 @@ export function DidYouMeanTest() {
+            test = Mock.optionsComponentSetup<DidYouMean, IDidYouMeanOptions>(DidYouMean, <IDidYouMeanOptions>{
+              enableAutoCorrection: true
+            });
+            test.env.queryStateModel.get = () => 'foobar';
           });
-          test.env.queryStateModel.get = () => 'foobar';
-          var qsmSpy = jasmine.createSpy('queryStateModelSpy');
-          test.env.queryStateModel.set = qsmSpy;
 
-          Simulate.query(test.env, {
-            results: FakeResults.createFakeResults(0),
-            queryCorrections: [fakeQueryCorrection]
-          });
+          it('should autocorrect the query when no results are found', () => {
+            const qsmSpy = jasmine.createSpy('queryStateModelSpy');
+            test.env.queryStateModel.set = qsmSpy;
 
-          expect(qsmSpy).toHaveBeenCalledWith(QueryStateModel.attributesEnum.q, 'this is the corrected query');
-        });
+            Simulate.query(test.env, {
+              results: FakeResults.createFakeResults(0),
+              queryCorrections: [fakeQueryCorrection]
+            });
 
-        it('set to true should send an analytics event when query is autocorrected', function() {
-          test = Mock.optionsComponentSetup<DidYouMean, IDidYouMeanOptions>(DidYouMean, <IDidYouMeanOptions>{
-            enableAutoCorrection: true
+            expect(qsmSpy).toHaveBeenCalledWith(QueryStateModel.attributesEnum.q, 'this is the corrected query');
           });
-          test.env.queryStateModel.get = () => 'foobar';
-          var analyticsSpy = jasmine.createSpy('analyticsSpy');
-          test.cmp.usageAnalytics.logSearchEvent = analyticsSpy;
 
-          Simulate.query(test.env, {
-            results: FakeResults.createFakeResults(0),
-            queryCorrections: [fakeQueryCorrection]
+          it('should send an analytics event when query is autocorrected', () => {
+            const analyticsSpy = jasmine.createSpy('analyticsSpy');
+            test.cmp.usageAnalytics.logSearchEvent = analyticsSpy;
+
+            Simulate.query(test.env, {
+              results: FakeResults.createFakeResults(0),
+              queryCorrections: [fakeQueryCorrection]
+            });
+
+            expect(analyticsSpy).toHaveBeenCalledWith(analyticsActionCauseList.didyoumeanAutomatic, {});
+            expect(analyticsSpy.calls.count()).toBe(1);
           });
 
-          expect(analyticsSpy).toHaveBeenCalledWith(analyticsActionCauseList.didyoumeanAutomatic, {});
-          expect(analyticsSpy.calls.count()).toBe(1);
+          it('should replace the state in history manager when query is autocorrected', () => {
+            spyOn(test.env.searchInterface.historyManager, 'replaceState');
+
+            Simulate.query(test.env, {
+              results: FakeResults.createFakeResults(0),
+              queryCorrections: [fakeQueryCorrection]
+            });
+
+            expect(test.env.searchInterface.historyManager.replaceState).toHaveBeenCalled();
```

Comment:
Small point, but I think it is better to use `toHaveBeenCalledTimes` for void methods as a more strict check than `toHaveBeenCalled` to make sure no unexpected calls are being made.